add_to_app/multiple_flutters/multiple_flutters_android/app/src/main/java/dev/flutter/multipleflutters/DataModel.kt

```diff
@@ -0,0 +1,41 @@
+package dev.flutter.multipleflutters
+
+import java.lang.ref.WeakReference
+
+/**
+ * Interface for getting notifications when the DataModel is updated.
+ */
+interface DataModelObserver {
+    fun onCountUpdate(newCount : Int)
+}
+
+/**
+ * A singleton/observable data model for the data shared between Flutter and the host platform.
+ *
+ * This is the definitive source of truth for all data.
+ */
+class DataModel {
+    companion object {
+        val instance = DataModel()
+    }
+
+    private val observers =  mutableListOf<WeakReference<DataModelObserver>>()
```

Comment:
Same comment as swift. Should this be a set? Since on Android it's not happening on an SDK class where the lifecycle is very clear, alternatively add some state assertions to the consuming class?

Reply:
It isn't necessary for this example to function correctly.  I think a designer of a system for production code would have to make the semantics very clear about what happens when you double subscribe the same object.  Another option is just to throw an exception if someone double subscribes.

Reply:
Either way is ok. I'd just add some comments or docs saying we're taking precautions with the engine bindings or this is a simple demo and you might run into trouble in these circumstances.